internal/provider/data_source_certificate.go

```diff
@@ -104,32 +112,51 @@ func dataSourceCertificate() *schema.Resource {
 	}
 }
 
-func dataSourceCertificateRead(d *schema.ResourceData, _ interface{}) error {
-	u, err := url.Parse(d.Get("url").(string))
+func dataSourceCertificateRead(d *schema.ResourceData, m interface{}) error {
+	config := m.(*providerConfig)
+
+	targetURL, err := url.Parse(d.Get("url").(string))
 	if err != nil {
 		return err
 	}
-	if u.Scheme != "https" {
-		return fmt.Errorf("invalid scheme")
-	}
-	if u.Port() == "" {
-		u.Host += ":443"
-	}
 
-	verifyChain := d.Get("verify_chain").(bool)
+	// Determine if we should verify the chain of certificates, or skip said verification
+	shouldVerifyChain := d.Get("verify_chain").(bool)
 
-	conn, err := tls.Dial("tcp", u.Host, &tls.Config{InsecureSkipVerify: !verifyChain})
+	// Ensure a port is set on the URL, or return an error
+	var peerCerts []*x509.Certificate
+	switch targetURL.Scheme {
+	case HTTPSScheme.String():
+		if targetURL.Port() == "" {
+			targetURL.Host += ":443"
+		}
+
+		// TODO remove this branch and default to use `fetchPeerCertificatesViaHTTPS`
+		//   as part of https://github.com/hashicorp/terraform-provider-tls/issues/183
+		if config.isProxyConfigured() {
```

Comment:
I still think the next major version should still perform this type of check and use the TLS client whenever possible for simplicity -- the check could see if `http.ProxyFromEnvironment` would actually use a proxy for the request by checking for `nil`.

Reply:
The reason I insist on using an actual HTTP client, is that we were and are requiring a `scheme` for the `url` argument. The fact that practitioners have to write `https://` is, in my opinion, an implicit implication we are making.

The original implementation simply didn't respect that. It would have made more sense if there was no scheme, and then it only (and exclusively) used `tls.Client`. Though that would have prevented even talking about supporting HTTP Proxies.